src/component.js

```diff
@@ -120,10 +120,13 @@ function renderComponent(component) {
 
 	if (parentDom) {
 		let commitQueue = [];
+		const nextVNode = assign({}, vnode);
+		nextVNode._original = nextVNode;
```

Comment:
reset this reference, if we don't `setState` would never execute because `_original` would still be `oldVNode`

Reply:
Any idea if it's cheaper/smaller to instead route this through `createVNode`? The assign here is going to create a VNode object with a different constructor function than all other vnodes.

Reply:
Oh I'll try and check that out

Reply:
The problem with that is that we're missing a lot of `vnode` properties if we call `createVNode`.